backend/messaging.py

```diff
@@ -34,35 +34,59 @@
 from .decorators import db_init_with_credentials
 from queue import LifoQueue
 
+class Messages(APIView):
+    queryset = None
+    serializer_class = MessageSerializer
+    def get(self, request, room_id, org_id):
+        """
+        Fetches all messages in a particular room of a particular organization.
 
-@swagger_auto_schema(
-    methods=["post", "get"],
-    query_serializer=GetMessageSerializer,
-    operation_summary="Creates and get messages",
-    responses={201: MessageResponse, 400: "Error: Bad Request"},
-)
-@sync_to_async
-@api_view(["GET", "POST"])
-@db_init_with_credentials
-def message_create_get(request, room_id):
-    if request.method == "GET":
+        Args:
+            request (dict): The request body usually containing the date
+            room_id (str): The id of the room where the request is being made
+            org_id (str): The id of the organization of the user
+
+        Returns:
+            list: Returns a list of available messages(dict objects)
+        
+         Raises:
+            204: No message in the specified room.
+            404: Room does not exist.
+            400: Bad Request.
+        """
+        # Set the page size for the response     
         paginator = PageNumberPagination()
         paginator.page_size = 20
         date = request.GET.get("date", None)
+
+        # Serialize the request data i.e convert it to json
         params_serializer = GetMessageSerializer(data=request.GET.dict())
+
+        # check if the serialized params are valid. If true, run request.
         if params_serializer.is_valid():
+
+            # Check to see if the room exists in the database
             room = DB.read_query("dm_rooms", query={"_id": room_id})
```

Comment:
Hi boss, you need to set the organization id to be equal to the org_id passed in the url, right now, DB.read_query will bs using the hardcoded org_id in db.py file. So before using DB.read_query, u can do sth like DB.organization_id = org_id

Reply:
Oh, true. @paisoncodes

Reply:
I already did that when I did data_storage. I'll just change the DB to data_storage

Reply:
I think we should keep DB object. It's already an instance, so we don't need to instantiate again before assigning the org_id to it. @paisoncodes

Reply:
@paisoncodes u didn't make the change here.

Reply:
It's done now... @man-of-mind

Reply:
@paisoncodes You didn't push your changes?
Seems it's still the same, there is no place you set the ord_id in the get method, u only did it for the post method.

Reply:
pushed. It's in the commit below